src/main/java/com/ldtteam/structurize/util/BlockUtils.java

```diff
@@ -220,6 +230,58 @@ else if (blockState.getBlock() == Blocks.BAMBOO_SAPLING)
         }
     }
 
+    /**
+     * For structure placement, check if two blocks are alike or if action has to be taken.
+     * @param blockState1 the first blockState.
+     * @param blockState2 the second blockState.
+     * @param notSolid the not solid condition.
+     * @param fancy if fancy paste.
+     * @return true if nothing has to be done.
+     */
+    public static boolean areBlockStatesEqual(final BlockState blockState1, final BlockState blockState2, final Predicate<BlockState> notSolid, final boolean fancy)
+    {
+        if (blockState1 == null || blockState2 == null)
+        {
+            return true;
+        }
+
+        final Block block1 = blockState2.getBlock();
+        final Block block2 = blockState1.getBlock();
+
+        if (fancy)
+        {
+            if (block1 == ModBlocks.blockSubstitution || blockState2.equals(blockState1))
+            {
+                return true;
+            }
+
+            if (block1 == ModBlocks.blockSolidSubstitution)
+            {
+                if (notSolid.test(blockState1))
+                {
+                    return true;
+                }
+            }
+
+            //list of things to only check block for.
+            //For the time being any flower pot is equal to each other.
+            if (block1 == Blocks.FLOWER_POT)
```

Comment:
pretty sure this would belong into the specialEqual predicate

Reply:
Good question, that's what I wondered about.